api/rest/restcore/issues_rest.php

```diff
@@ -188,4 +192,32 @@ function rest_issue_note_delete( \Slim\Http\Request $p_request, \Slim\Http\Respo
 	$t_issue = mc_issue_get( /* username */ '', /* password */ '', $t_issue_id );
 	return $p_response->withStatus( HTTP_STATUS_SUCCESS, 'Issue Note Deleted' )->
 		withJson( array( 'issue' => $t_issue ) );
-}
\ No newline at end of file
+}
+
+/**
+ * Update an issue from a PATCH to the issues url.
+ *
+ * @param \Slim\Http\Request $p_request   The request.
+ * @param \Slim\Http\Response $p_response The response.
+ * @param array $p_args Arguments
+ * @return \Slim\Http\Response The augmented response.
+ */
+function rest_issue_update( \Slim\Http\Request $p_request, \Slim\Http\Response $p_response, array $p_args ) {
+	$t_issue = (object)$p_request->getParsedBody();
+
+	$t_issue_id = isset( $p_args['id'] ) ? $p_args['id'] : $p_request->getParam( 'id' );
+	if( is_blank( $t_issue_id ) ) {
+		$t_message = "Mandatory field 'id' is missing.";
+		return $p_response->withStatus( HTTP_STATUS_BAD_REQUEST, $t_message );
+	}
+
```

Comment:
You should call mc_issue_get() here, and merge the contents of the post body on top of the returned issue.  Hence, enabling the caller to just call PATCH with just the modified fields (e.g. handler and status).  This way the client doesn't need to get the full issue and then call patch with it.  This also behaves better if two clients are trying to to do update at the same time (e.g. one client updating handler + status and the other trying to change the priority.

Reply:
@vboctor when I get an issue, edit some fields and update the issue (RESP API), only the edited fields appear as modified in the Issue Change History. This merge is not executed in the mc_issue_update? And, what about the mandatory fields as summary, description or category?